fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/chartofaccounts/ChartOfAccountsImportHandler.java

```diff
@@ -0,0 +1,154 @@
+        for (int rowIndex=1;rowIndex<noOfEntries;rowIndex++){
+            Row row;
+            try {
+                row=chartOfAccountsSheet.getRow(rowIndex);
+                if (isNotImported(row,STATUS_COL)){
+                    glAccounts.add(readGlAccounts(row));
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private GLAccountData readGlAccounts(Row row) {
+        String accountType=readAsString(ACCOUNT_TYPE_COL,row);
+        Long accountTypeId=null;
+        EnumOptionData accountTypeEnum=null;
+        if (accountType!=null && accountType.equalsIgnoreCase("ASSET")){
+            accountTypeId=1L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }else if(accountType!=null && accountType.equalsIgnoreCase("LIABILITY")){
+            accountTypeId=2L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }else if(accountType!=null && accountType.equalsIgnoreCase("EQUITY")){
+            accountTypeId=3L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }
+        else if(accountType!=null && accountType.equalsIgnoreCase("INCOME")){
+            accountTypeId=4L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }else if(accountType!=null && accountType.equalsIgnoreCase("EXPENSE")){
+            accountTypeId=5L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }
+        String accountName=readAsString(ACCOUNT_NAME_COL,row);
+        String usage=readAsString(ACCOUNT_USAGE_COL,row);
+        Long usageId=null;
+        EnumOptionData usageEnum=null;
+        if (usage!=null&& usage.equals("Detail")){
+            usageId=1L;
+            usageEnum=new EnumOptionData(usageId,null,null);
+        }else if (usage!=null&&usage.equals("Header")){
+            usageId=2L;
+            usageEnum=new EnumOptionData(usageId,null,null);
+        }
+        Boolean manualEntriesAllowed=readAsBoolean(MANUAL_ENTRIES_ALLOWED_COL,row);
+        Long parentId=Long.parseLong(readAsString(PARENT_ID_COL,row));
+        String glCode=readAsString(GL_CODE_COL,row);
+        Long tagId=Long.parseLong(readAsString(TAG_ID_COL,row));
+        CodeValueData tagIdCodeValueData=new CodeValueData(tagId);
+        String description=readAsString(DESCRIPTION_COL,row);
+        return new GLAccountData(accountName,parentId,glCode,manualEntriesAllowed,accountTypeEnum,usageEnum,description,tagIdCodeValueData,row.getRowNum());
+    }
+
+    @Override
+    public void Upload(PortfolioCommandSourceWritePlatformService commandsSourceWritePlatformService) {
+        Sheet chartOfAccountsSheet=workbook.getSheet("ChartOfAccounts");
+        for (GLAccountData glAccount: glAccounts) {
+            try {
+                GsonBuilder gsonBuilder = new GsonBuilder();
```

Comment:
creation of gsonBuilder and registering type adapter can be done out of loop only once. inside loop just pass toJson(glAccount);